### PR TITLE
Added add price modal and various bug fixes

### PR DESCRIPTION
**Pull request by UnluckyBird**
Added add price modal
Fixed routing bug when group was found
Fixed background popping through on swipable start Fixed modals opening again immediately after being closed
Fixed modals opening again immediately after being closed

### CODE REVIEW

**Review by andreasalstrup — APPROVED** [review]
LGTM

**Review by mikejensen0 — APPROVED** [review]
LGTM, but I still think opening the edit list item modal takes a bit too long

File: app/(tabs)/shoppingList/index.tsx [hunk 1]
```diff
@@ -188,21 +212,23 @@ export default function ToBeBoughtScreen() {
...
-          activeOpacity={0.5}>
-          <GestureDetector gesture={Gesture.LongPress().onStart(e => {
-            setItemToEdit(index)
-            editProduct(index)
-            handleModalAddOrEditItem()
-          })}> 
-            <View style={[styles.container, {backgroundColor: index % 2 == 0 ? Colors[colorScheme].listBackgroundColor1 : Colors[colorScheme].listBackgroundColor2}]}>
-              <View style={styles.item}>
-                <Text style={styles.itemText}>{item.name}</Text>      
+        <GestureDetector gesture={Gesture.LongPress().onStart(e => {
```
[the remark this answers is not in the document]
Comment (UnluckyBird) [reply app/(tabs)/shoppingList/index.tsx:215]: It uses the standard getLongPressTimeout, that is defined by Android: https://developer.android.com/reference/android/view/ViewConfiguration#getLongPressTimeout()

Don't know that I agree that we should change

[the remark this answers is not in the document]
Comment (UnluckyBird) [reply app/(tabs)/shoppingList/index.tsx:215]: Changed it from 500ms to 300ms

**Pull request merged**